### PR TITLE
chore: updating supported python versions

### PR DESCRIPTION
**Pull request by Artui**
Addresses #28 issue

### CODE REVIEW

**Comment by kvz** [comment]
Sadly the build failed. I’m not deep into python and could signal a maintainer but perhaps you can see what’s the trouble?

perhaps the version was limited for a good reason 🤔

**Comment by Artui** [comment]
> Sadly the build failed. I’m not deep into python and could signal a maintainer but perhaps you can see what’s the trouble?
> 
> perhaps the version was limited for a good reason 🤔

One of the dependencies with the current version works for python 3.7, but doesn't work for 3.12. If I update it to the version that supports 3.12, we lose support for 3.7 :D 
I wonder if we're okay with losing support for 3.7?

**Comment by Acconut** [comment]
We're fine with dropping support for older Python release, but that has to part of a major release as it's a breaking change. I don't see any reason against a major release, but we might want to use this opportunity to also drop 3.8, whose support ends in November 2024: https://devguide.python.org/versions/

**Comment by Artui** [comment]
> We're fine with dropping support for older Python release, but that has to part of a major release as it's a breaking change. I don't see any reason against a major release, but we might want to use this opportunity to also drop 3.8, whose support ends in November 2024: https://devguide.python.org/versions/

I'm down for that! Added this to the PR :)

**Comment by Artui** [comment]
Also updated version to 0.3.0, not sure if we want to do 1.x.x, since the version was 0.2.2 before my PR

**Comment by kvz** [comment]
When cutting major releases, I think we have to involve the experts, @arnaudlimbourg & @ifedapoolarewaj.

do we see other opportunities for upgrades that are breaking, that we should be making in this release?

**Comment by Acconut** [comment]
You misspelled @ifedapoolarewaju :)

**Comment by Acconut** [comment]
> Also updated version to 0.3.0, not sure if we want to do 1.x.x, since the version was 0.2.2 before my PR

If we do a major release, it will be 1.0.0

**Comment by Artui** [comment]
> > Also updated version to 0.3.0, not sure if we want to do 1.x.x, since the version was 0.2.2 before my PR
> 
> If we do a major release, it will be 1.0.0

Sure, it's just been in 0.x.x for a while, just wanted to check in if it's ready for 1.0.0

**Comment by Artui** [comment]
Any updates on the status here? @kvz @arnaudlimbourg @ifedapoolarewaju

**Comment by ifedapoolarewaju** [comment]
Hi, I just realised that we need to update the [__version__](https://github.com/transloadit/python-sdk/blob/main/transloadit/__init__.py#L1) variable as well. Otherwise the SDK will be sending the wrong version to Transloadit servers.

When we do that, we'd need to update this line in [the test](https://github.com/transloadit/python-sdk/blob/main/tests/test_request.py#L22) so it passes

**Comment by Artui** [comment]
I see that all the tests have passed now, thanks for approving the pipeline. How long would it take to get a release out there? I don't see a PyPi deployment piece in the Github Actions code, so I assume it's a manual process done by one of the maintainers?

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pytransloadit"
-version = "0.2.2"
+version = "0.3.0"
```
Comment (ifedapoolarewaju) [inline pyproject.toml:3]: I guess this needs to be 1.0 as well then
```suggestion
version = "1.0.0"
```

**Review by ifedapoolarewaju — APPROVED** [review]

**Comment by ifedapoolarewaju** [comment]
This looks all good to me.
@kvz @Acconut would there be any other objections to merging this?

**Review by Acconut — APPROVED** [review]
Looks good, thanks for the feedback!

**Comment by Artui** [comment]
Awesome, thanks!

**Comment by Acconut** [comment]
One more thing: Can you please update the README? It currently says:

> Only Python 3.7+ versions are supported.

**Comment by Artui** [comment]
@Acconut I've updated that and updated CHANGELOG as well

**Review by Acconut — APPROVED** [review]
@cdr-chakotay Could you merge and release this PR?

**Comment by cdr-chakotay** [comment]
> @cdr-chakotay Could you merge and release this PR?

Sure, we have had some internal discussions on further release. Will come back to you :)

**Comment by cdr-chakotay** [comment]
@Artui Merged this one and will release soon. Thanks for your contribution!

**Pull request merged**